src/metaArray.h

```diff
@@ -54,160 +55,186 @@ class METAIO_EXPORT MetaArray : public MetaForm
-    void         AutoFreeElementData(bool _freeData);
-
-    const char * ElementDataFileName(void) const;
-    void         ElementDataFileName(const char * _dataFileName);
-
-    void *       ElementData(void);
-    double       ElementData(int _i) const;
-    void         ElementData(void * _data, bool _autoFreeElementData=false);
-    bool         ElementData(int _i, double _v);
-
-    //
-    //
-    //
-    virtual bool CanRead(const char *_headerName=nullptr) const;
-
-    virtual bool Read(const char *_headerName=nullptr,
-                      bool _readElements=true,
-                      void * _elementDataBuffer=nullptr,
-                      bool _autoFreeElementData=false);
-
-    virtual bool CanReadStream(std::ifstream * _stream) const;
-
-    virtual bool ReadStream(std::ifstream * _stream,
-                            bool _readElements=true,
-                            void * _elementDataBuffer=nullptr,
-                            bool _autoFreeElementData=false);
-
-    virtual bool Write(const char *_headName=nullptr,
-                       const char *_dataName=nullptr,
-                       bool _writeElements=true,
-                       const void * _constElementData=nullptr);
+public:
+  ////
+  //
+  // Constructors & Destructor
+  //
+  ////
+  MetaArray(void);
+
+  MetaArray(const char * _headerName);
+
+  MetaArray(MetaArray * _vector, bool _allocateElementData = false, bool _autoFreeElementData = false);
+
+  MetaArray(int               _length,
+            MET_ValueEnumType _elementType,
+            int               _elementNumberOfChannels = 1,
+            void *            _elementData = nullptr,
+            bool              _allocateElementData = false,
+            bool              _autoFreeElementData = false);
+
+  ~MetaArray(void) override;
+
+  void
+  PrintInfo(void) const override;
 
-    virtual bool WriteStream(std::ofstream * _stream,
-                             bool _writeElements=true,
-                             const void * _constElementData=nullptr);
+  void
+  CopyInfo(const MetaForm * _form) override;
+
+  void
+  Clear(void) override;
+
+  bool
+  InitializeEssential(int               _nDims,
+                      MET_ValueEnumType _elementType,
+                      int               _elementNumberOfChannels = 1,
+                      void *            _elementData = nullptr,
+                      bool              _allocateElementData = false,
+                      bool              _autoFreeElementData = true);
+
+  bool
+  AllocateElementData(bool _autoFreeElementData = true);
+
+  int
+  Length(void) const;
+  void
+  Length(int _length);
+
+  int
+  NDims(void) const;
+  void
+  NDims(int _length);
+
+  MET_ValueEnumType
+  ElementType(void) const;
+  void
+  ElementType(MET_ValueEnumType _elementType);
+
+  int
+  ElementNumberOfChannels(void) const;
+  void
+  ElementNumberOfChannels(int _elementNumberOfChannels);
+
+  //
```

Comment:
Is there a chance to remove these annoying empty comments during this style update?

Reply:
Formatters generally do not touch comments (except to line wrap).

Reply:
Removal of the comments can be an additional commit.